opentelemetry-sdk/test-requirements.txt

```diff
@@ -0,0 +1,19 @@
+asgiref==3.7.2
+attrs==23.2.0
+Deprecated==1.2.14
+flaky==3.7.0
```

Comment:
Still not super convinced that we need to specify indirect dependencies and their versions (I understand that the original purpose is to prevent random tests not passing based off of bugs introduced in other libraries) but we can try this out for now to see if it improves developer experience.